### PR TITLE
MINOR: Update build and test dependencies for 3.5

### PR DESCRIPTION
**Pull request by ijuma**
The most recent version that didn't cause compiler or test errors was chosen.

Gradle plugin updates follow, they include minor fixes and support
for newer Java and Gradle versions:
* spotless: 6.10.0 -> 6.13.0
* versions: 0.42.0 -> 0.44.0
* dependencycheck: 7.1.1 -> 8.0.2
* rat: 0.7.1 -> 0.8.0
* spotbugs: 5.0.9 -> 5.0.13
* test-retry: 1.4.0 -> 1.5.1
* scoverage: 7.0.0 -> 7.0.1

Test/benchmark updates:
* jmh: 1.35 -> 1.36
  GC profiler and perfasm improvements
  https://mail.openjdk.org/pipermail/jmh-dev/2022-November/003553.html
* junit: 5.9.0 -> 5.9.2
  Minor fixes
  https://junit.org/junit5/docs/current/release-notes/index.html#release-notes-5.9.2
* jqwik: 1.6.5 -> 1.7.2
  A number of usability improvements, bug fixes and minor breaking changes
  https://jqwik.net/release-notes.html#172
* mockito: 4.6.1 -> 4.9.0
  Usability improvements, perf fixes and bug fixes
  https://github.com/mockito/mockito/releases/tag/v4.7.0
  https://github.com/mockito/mockito/releases/tag/v4.8.0
  https://github.com/mockito/mockito/releases/tag/v4.8.1
  https://github.com/mockito/mockito/releases/tag/v4.9.0

### Committer Checklist (excluded from commit message)
- [ ] Verify design and implementation 
- [ ] Verify test coverage and CI build status
- [ ] Verify documentation (including upgrade notes)

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -30,16 +30,16 @@ buildscript {
 }
 
 plugins {
-  id 'com.diffplug.spotless' version '6.10.0'
-  id 'com.github.ben-manes.versions' version '0.42.0'
+  id 'com.diffplug.spotless' version '6.14.1'
```
Comment (showuon) [inline build.gradle]: The latest version of `spotless-plugin-gradle` in maven is `6.14.0`.
ref: https://mvnrepository.com/artifact/com.diffplug.spotless/spotless-plugin-gradle

Comment (ijuma) [reply build.gradle]: 6.14.1 is available, see here https://search.maven.org/artifact/com.diffplug.spotless/spotless-plugin-gradle/6.14.1/jar

But it requires Java 11 to build, so I had to downgrade to 6.13.0..

I'll ping reviewers once I get a green build. :)

**Comment by clolov** [comment]
Do we use some tool which suggests these newer versions or we do the checks manually for each release?

**Comment by ijuma** [comment]
I used `gradlew dependencyUpdates`. A tool that automatically retrieved the release notes/changelogs would be useful. That is the more time consuming part vs the version bumps.

**Comment by ijuma** [comment]
JDK 11 build passed, the other two had unrelated failures:

>  Build / JDK 17 and Scala 2.13 / org.apache.kafka.common.network.SslTransportLayerTest.[2] tlsProtocol=TLSv1.2, useInlinePem=true	20 sec	1
>  Build / JDK 8 and Scala 2.12 / kafka.admin.ReassignPartitionsIntegrationTest.testProduceAndConsumeWithReassignmentInProgress(String).quorum=kraft

**Comment by ijuma** [comment]
@showuon This is now ready for review.

**Review by stanislavkozlovski — APPROVED** [review]
LGTM! Thanks for the PR.

**Review by showuon — APPROVED** [review]
LGTM!

Failed tests are unrelated
```
    Build / JDK 17 and Scala 2.13 / org.apache.kafka.common.network.SslTransportLayerTest.[2] tlsProtocol=TLSv1.2, useInlinePem=true
    Build / JDK 8 and Scala 2.12 / kafka.admin.ReassignPartitionsIntegrationTest.testProduceAndConsumeWithReassignmentInProgress(String).quorum=kraft
```

**Pull request merged**